pallets/liquid-staking/src/lib.rs

```diff
@@ -871,6 +892,10 @@ pub mod pallet {
                     Self::verify_merkle_proof(key, value, proof),
                     Error::<T>::InvalidProof
                 );
+                let rewards = staking_ledger.total.saturating_sub(ledger.total);
+                if let Err(e) = Self::do_charge_commission(rewards) {
```

Comment:
can we seperate `do_charge_commission` into two parts which may look better:

- get_inflation_amount
- mint